app/models/concerns/foreman_openscap/host_extensions.rb

```diff
@@ -109,15 +101,13 @@ def search_by_policy_name(key, operator, policy_name)
 
       def search_by_missing_arf(key, operator, policy_name)
         cond = sanitize_sql_for_conditions(["foreman_openscap_policies.name #{operator} ?", value_to_sql(operator, policy_name)])
-        { :conditions => Host::Managed.arel_table[:id].in(Host::Managed.select(Host::Managed.arel_table[:id])
-            .joins(:policies)
-            .where(cond)
-            .where("foreman_openscap_assets.id NOT IN (
-                     SELECT DISTINCT foreman_openscap_arf_reports.asset_id
-                     FROM foreman_openscap_arf_reports
-                     WHERE foreman_openscap_arf_reports.asset_id = foreman_openscap_assets.id
-                         AND foreman_openscap_arf_reports.policy_id = foreman_openscap_policies.id)")
-            .pluck(:id)).to_sql}
+        host_ids_from_arf_of_policy = ForemanOpenscap::ArfReport.joins(:policy).where(cond).pluck(:host_id).uniq
```

Comment:
It was working when I tested, the policy is never connected directly to the host - it is linked through Asset or ArfReport. It does not matter if a certain policy comes from a hostgroup - it is still linked to a host through associated asset.